enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,441 @@
+last-updated: 2022-01-17
+tracking-link:
+  - https://issues.redhat.com/browse/RFE-2240
+  - https://issues.redhat.com/browse/OCPPLAN-4577
+status: implementable
+---
+
+# heterogeneous-architecture-cluster-support
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for tech preview and GA
+- [ ] User-facing documentation is created in [Openshift-docs](https://github.com/Openshift/Openshift-docs/)
+
+## Summary
+
+This enhancement describes the plan to support provisioning and upgrades of heterogeneous architecture Openshift clusters.
+A heterogeneous cluster is a cluster with nodes running on different architectures with a homogeneous control plane and worker machinesets
+supporting any combination of architectures (eg: x86 control plane + a mix of ARM/P/Z worker nodes). This document describes the support
+needed for heterogeneous architecture clusters in phases. There will be three major phases with basic support in phase 1 which is targeted
+for a 4.11 techpreview (with documented limitations) and the other phases which address gaps beyond the 4.11 time frame. The targeted platforms
+in the initial phases are Azure and AWS with support for x86 and ARM architectures.
+
+![alt heterogeneous cluster](heterogeneous-cluster.png)
+
+## Motivation
+
+- Users can take advantage of cost savings on architectures like ARM.
+- Users can colocate applications that require different architectures rather than needing multiple clusters and thus avoid cross cluster
+latency and connectivity failures.
+
+### Goals
+
+- Provide a way to install and upgrade clusters with varied architecture machinesets with the ability to autoscale.
+- Support provisioning heterogeneous architecture clusters in Hypershift.
+- Generate a manifestlist based payload that contains manifestlist references for the 4 architectures OpenShift supports (x86, ARM, Power and Z).
+
+### Non-Goals
+
+- Building manifestlisted payloads using Openshift builds (Note: our CI payload build infrastructure is based on builds, so there will be no CI-built
+heterogeneous payload+CI testing.  There are also no non-x86 CI payloads from which to construct a heterogeneous CI payload. Heterogeneous nightly
+payloads will be built+subjected to CI test jobs in the same way as homogeneous nightlies).
+- Heterogeneous control plane.
+- Heterogeneous to homogeneous cluster migration support
+
+## Proposal
+
+The support for installation and upgrade of a heterogeneous architecture cluster and the associated components will be done in phases. The phases
+range from a bare bones no upgrade support installation to the full fledged installation and upgrade capability of a cluster on different topologies.
+The phases are listed below and details on some components of the phases are described below.
+
+Phase 0: Prototype
```

Comment:
What phase covers disconnected installations? (with all the mirroring, including olm, acm...)

Reply:
the support for oc adm mirror is being worked on so this will be available in 4.12.